### PR TITLE
Enhanced/improved provider configuration options and documentation

### LINKED ISSUE #170: Use kube context - as `argocd --core`
Hi,
I have not exposed argo server, it is as ClusterIP. I'd like to connect to it using kubernetes api like `argocd --core` does
```
$ argocd version --core
argocd: v2.3.3+07ac038
  BuildDate: 2022-03-30T01:46:59Z
  GitCommit: 07ac038a8f97a93b401e824550f0505400a8c84e
  GitTreeState: clean
  GoVersion: go1.17.6
  Compiler: gc
  Platform: linux/amd64
argocd-server: v2.3.3+07ac038
  BuildDate: 2022-03-30T01:46:59Z
  GitCommit: 07ac038a8f97a93b401e824550f0505400a8c84e
  GitTreeState: clean
  GoVersion: go1.17.6
  Compiler: gc
  Platform: linux/amd64
  Ksonnet Version: unable to determine ksonnet version: exec: "ks": executable file not found in $PATH
  Kustomize Version: could not get kustomize version: exec: "kustomize": executable file not found in $PATH
  Helm Version: v3.8.2+g6e3701e
  Kubectl Version: v0.23.1
  Jsonnet Version: v0.18.0
```

How to configure this provider? 
```yaml
provider "argocd" {
  # Configuration options
  # server_addr = "kubernetes"
  insecure = true
  kubernetes {
    config_path = "~/.kube/config"
    config_context = "kind-monitor"
  }
}
```
`server_addr` attribure is required :( but if I set it with kubectl api it does't work.
```
Error: Missing required argument
│ 
│   with provider["registry.terraform.io/oboukili/argocd"],
│   on providers.tf line 10, in provider "argocd":
│   10: provider "argocd" {
│ 
│ "username": one of `auth_token,password,use_local_config,username` must be specified
```

### LINKED ISSUE #202: Terraform block
I would like to connect argocd cluster with terraform but  i cannot use kubernetes block due to " server_addr " is required. I would like to use kubernetes block like below but i got " kubernetes block not expected here" error. Maybe i am using the kubernetes block the wrong way. Could you please help me about this?

```tf
provider "argocd" {
  context = "../../../../.kube/config"
  kubernetes {
    host                   = var.endpoint
    cluster_ca_certificate = var.kubeconfig-certificate-authority-data
    exec {
        api_version = "client.authentication.k8s.io/v1beta1"
        command     = "aws"
        args = [
            "eks",
            "get-token",
            "--cluster-name",
            var.cluster_name
            ]
        }
    }
  server_addr = "argocd.${data.local_file.argocd_server_addr.content}.cluster.local:443"  # env ARGOCD_SERVER input
  #auth_token  = "1234..."          # env ARGOCD_AUTH_TOKEN
  username  = "admin"            # env ARGOCD_AUTH_USERNAME
  password  = data.local_file.argocd_server_password.content             # env ARGOCD_AUTH_PASSWORD
  insecure  = true              # env ARGOCD_INSECURE
}
```

### LINKED ISSUE #267: Provider authentication details/examples?
### Terraform Version, ArgoCD Provider Version and ArgoCD Version
```
Terraform version: 1.4.5
ArgoCD provider version: none yet
ArgoCD version: 2.6.7
```

## Terraform configuration
None yet

## Question
I'm adopting argo and looking to plan my authentication path for the provider, but the documentation isn't really clear on which sets of options go together by way of examples.

For instance, I know that I could authenticate with a token or user/pass, and that will probably satisfy the constraints.  But if I authenticate with `kubernetes` (similarly to the `kubernetes` or `helm` providers) do I still need to auth to argo separately, or is a kube administrator enough?  

I can make a machine user in argo and store a token and go through a public access point to argo, but I'm trying to limit authentication to my oauth provider so I don't want to do that unless it's necessary (hence asking about the kubernetes auth path).

### PR DESCRIPTION
**Pull request by onematchfox**
It seems that lots of folks struggle with how to configure the provider, so I decided to try better understand the various configuration options myself and improve the documentation around this. 

In doing so, I spotted a number of configuration options on the `kubernetes` block that were never used. So, I implemented those that could be implemented and removed those that could not be implemented. 

Additionally, while delving through the argocd CLI code, I found that we can actually implement support for ArgoCD core, albeit with some restrictions/quid-pro-quos (as documented). 

The outcome of all of this is, hopefully, far better documentation on the ways in which the provider can be configured (alongside examples of the varying options). 

Closes #267
Closes #202
Closes #170

Partial resolution of #135

### CODE REVIEW

**Review by oboukili — CHANGES_REQUESTED** [review]
Great !

**Comment by onematchfox** [comment]
@oboukili, quick reminder that I'm waiting on a final review here.

**Review by oboukili — APPROVED** [review]

**Pull request merged**
